core/src/main/java/com/klaytn/caver/wallet/keyring/Keyring.java

```diff
@@ -0,0 +1,741 @@
+     * @return MessageSigned
+     */
+    public MessageSigned signMessage(String message, int roleIndex, int keyIndex) {
+        PrivateKey[] groupKeyArr = getKeyByRole(roleIndex);
+        if(keyIndex < 0) throw new IllegalArgumentException("keyIndex cannot have negative value.");
+        if(keyIndex >= groupKeyArr.length) throw new IllegalArgumentException("keyIndex value must be less than the length of key array");
+
+        String messageHash = Utils.signMessage(message);
+        KlaySignatureData signatureData = groupKeyArr[keyIndex].signMessage(messageHash);
+        return new MessageSigned(messageHash, signatureData, message);
+    }
+
+    /**
+     * Returns keys corresponding to the role index.
+     * If the keys corresponding to the role index is empty, the default key is returned.(Role index 0)
+     * @param roleIndex A number indicating the role of the key.
+     * @return An array of PrivateKey
+     */
+    public PrivateKey[] getKeyByRole(int roleIndex) {
+        if(roleIndex >= AccountKeyRoleBased.MAX_ROLE_BASED_KEY_COUNT) {
+            throw new IllegalArgumentException("Invalid role index");
+        }
+
+        PrivateKey[] groupKeyArr = this.keys.get(roleIndex);
+
+        if(groupKeyArr.length == 0 && roleIndex > AccountKeyRoleBased.RoleGroup.TRANSACTION.getIndex()) {
+            groupKeyArr = this.keys.get(AccountKeyRoleBased.RoleGroup.TRANSACTION.getIndex());
+
+            if(groupKeyArr.length == 0) {
+                throw new RuntimeException("The key data with specified roleIndex does not exist. The default key in TransactionRole is also empty.");
+            }
+        }
+
+        return groupKeyArr;
+    }
+
+    /**
+     * Returns a KlaytnWalletKey format
+     * @return String
+     */
+    public String getKlaytnWalletKey() {
+        String errorMessage = "The keyring cannot be exported in KlaytnWalletKey format. Use caver.wallet.keyring.encrypt or keyring.encrypt.";
+
+        PrivateKey[] txRoleGroupKeyArr = this.keys.get(AccountKeyRoleBased.RoleGroup.TRANSACTION.getIndex());
+        if(txRoleGroupKeyArr.length != 1) {
+            throw new RuntimeException(errorMessage);
+        }
+
+        for(int i = AccountKeyRoleBased.RoleGroup.ACCOUNT_UPDATE.getIndex(); i<AccountKeyRoleBased.MAX_ROLE_BASED_KEY_COUNT; i++) {
+            if(this.keys.get(i).length > 0) {
+                throw new RuntimeException(errorMessage);
+            }
+        }
+
+        String address = Numeric.prependHexPrefix(this.address);
+        String privateKeyStr = Numeric.prependHexPrefix(txRoleGroupKeyArr[0].getPrivateKey());
+
+        return privateKeyStr + "0x00" + address;
+    }
+
+    /**
+     * Returns an instance of Account(AccountKeyPublic).
+     * @return Account
+     */
+    public Account toAccount() {
+        boolean isExistsOtherGroupKeys = this.keys.stream()
+                .skip(1)
+                .anyMatch(groupKeyArr -> groupKeyArr.length != 0);
+
+        if(isExistsOtherGroupKeys) {
```

Comment:
If user call keyring.toAccount() without options and keyring contains more than one PrivateKey inside, toAccount create default WeightedMultiSigOptions or WeightedMultiSigOptions[] to make an Account instance.

Reply:
ok. I'll add this feature